src/aws/osml/model_runner/sink/sink_factory.py

```diff
@@ -12,7 +13,7 @@
 logger = logging.getLogger(__name__)
 
 
-class SinkFactory(Sink):
+class SinkFactory(Sink, ABC):
```

Comment:
SinkFactory should not extend Sink or ABC. It only needs to extend ABC if it will implement abstract methods that must be implemented by subclasses to use the factory. It shouldn't implement Sink because it isn't actually providing the name(), mode(), and write(), functions that define the Sink interface. I realize that extending Sink was in the base code not something you added but this change drew my attention to it.

Reply:
I have removed the extension of ABC and Sink in the base class.